src/tls13.c

```diff
@@ -2897,9 +2897,9 @@ static int SetupPskKey(WOLFSSL* ssl, PreSharedKey* psk, int clientHello)
     }
 #endif
 
-    if (ssl->options.noPskDheKe) {
-        ssl->arrays->preMasterSz = 0;
-    }
+    /* pre master buffer is still allocated and size set, since this client PSK
```

Comment:
We should only get here if we are doing the PSK.

Reply:
Correct. Client is trying to resume with TLS v1.3 session ticket and server opts to do a new handshake and sends new keyshare in server hello. The big was this caused EccSharedSecret to be called with a zero size for the master secret.

Reply:
Then the problem is that it is processing the KeyShare when we have chosen not to (noPskDheKe).
The change to KeyShare extension code is that it needs skips the creation of the secret if we resuming and noPskDheKe is set. Still want to parse it do check for valid extension.

Reply:
Fixed. The server was not honoring the `psk_de` requirement from the client if the PSK resumption failed.